pkg/cli/admin/waitfornoderestart/wait_for_node_restart.go

```diff
@@ -0,0 +1,281 @@
+package waitfornoderestart
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/openshift/oc/pkg/cli/admin/rebootmachineconfigpool"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	nodeKind = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Node"}
+)
+
+const (
+	RemoveOldTrustFieldManager = "remove-old-trust"
+)
+
+type WaitForNodeRestartRuntime struct {
+	ResourceFinder genericclioptions.ResourceFinder
+	KubeClient     kubernetes.Interface
+	DynamicClient  dynamic.Interface
+
+	RebootNumber int
+
+	masterRestartNumber int
+	workerRestartNumber int
+
+	genericclioptions.IOStreams
+}
+
+func (r *WaitForNodeRestartRuntime) setRestartNumber(ctx context.Context) error {
+	if r.RebootNumber > 0 {
```

Comment:
Hmm, not sure how useful this is as a general case, but I guess the expectation is that all nodes will be rebooting in tandem?

As in, we aren't really expecting users to run this command 10x for worker pools just because they feel like it for other use cases.

Reply:
If I were a cluster-admin trying to restart a particular pool, and knew about this, I would create a machineconfig that overrides the file with my own value and then run this command with a specific, "expect this one" against the nodes.